suite-common/fiat-services/src/coingecko.ts

```diff
@@ -103,39 +103,57 @@ export const fetchCurrentFiatRates = async (ticker: TickerId) => {
 };
 
 /**
- * Returns the historical rate for a given symbol, timestamp fetched from CoinGecko API.
- * Be aware that the data granularity is 1 day.
- * Returns null if coin for a given symbol was not found.
+ * Helper function that goes through timestamped fiat rates returned from Coingecko and finds the closest one to the provided timestamp.
+ * @returns [timestamp, fiatRate] pair
+ */
+const findClosestTimestampValue = (timestamp: number, arr: Array<[number, number]>): number => {
+    let closestTimestamp = arr[0];
+    for (let i = 1; i < arr.length; i++) {
+        const currentTimeDelta = Math.abs(timestamp - closestTimestamp[0] / 1000);
+        const nextTimeDelta = Math.abs(timestamp - arr[i][0] / 1000);
+
+        closestTimestamp = arr[i];
+
+        // The timestamps are ordered, if next time delta is higher, we can stop the iteration.
+        if (currentTimeDelta < nextTimeDelta) {
+            break;
+        }
+    }
+
+    return closestTimestamp[1];
+};
+
+/**
+ * Returns the historical rates for a given symbol adn array of timestamps, fetched from CoinGecko API.
```

Comment:
typo, I recently installed a spell checker in VS Code and I don't understand why I didn't do it much earlier 😄 
```suggestion
 * Returns the historical rates for a given symbol and array of timestamps, fetched from CoinGecko API.
```

Reply:
fixed in 6d30b1734